backend/src/data/repositories/user-repository.ts

```diff
@@ -1,23 +1,52 @@
-import { UserModel } from '../models';
+import { UserModel, DoctorModel, ClinicModel } from '../models';
 import { IUser } from '~/common/interfaces';
+import { UserType } from '~/common/enums';
+import { ModelAlias } from '~/common/enums';
 
 class UserRepository {
   public getAll():Promise<IUser[]>{
     return UserModel.findAll()
   }
+
+  public getByType(type:UserType):Promise<IUser[]>{
+    if (type === UserType.DOCTOR) {
+      return UserModel.findAll({
+        where: {type},
+        include: [
+          {
+            model: DoctorModel,
+            as: ModelAlias.DOCTORS,
+            attributes: ['id', 'department', 'about'],
+            include: [
+              {
+                model: ClinicModel,
+                as: ModelAlias.CLINICS,
+                attributes: ['id', 'name', 'address', 'clinicType']
```

Comment:
```suggestion
            attributes: ['id', 'department', 'about'],
            include: [
              {
                model: ClinicModel,
                as: ModelAlias.CLINICS,
                attributes: ['id', 'name', 'address', 'clinicType']
```
You should use enum for keys, instead just strings